### DISCUSSION TITLE
holes in drawing

### DISCUSSION THREAD

**Discussion by Timerlen** [mozman/ezdxf#375] [Q&A]
I was wondering, I don't know if it is right or it is possible to count the number of holes in this drawing 
![1](https://user-images.githubusercontent.com/80266841/112532766-80832580-8da9-11eb-9c59-a9e657a936f2.jpg)
can I do everything on SPLINE and check if SPLINE is closed and count? Maybe there is another method to solve my problem?

**Comment by mozman** [comment]
The SPLINE entity has a CLOSED flag (`Spline.closed` property), which could be checked, but as often this data is not reliable, the better way is to calculate the start- and end point and check if they are close.

```Python
for entity in msp.query("SPLINE"):
    bspline = entity.construction_tool()
    start, end = bspline.points([0, bspline.max_t])
    if start.isclose(end):
        # SPLINE is closed
        ...
```
